scipy/special/tests/test_basic.py

```diff
@@ -1666,8 +1666,9 @@ def test_euler(self):
         eu0 = special.euler(0)
         eu1 = special.euler(1)
         eu2 = special.euler(2)   # just checking segfaults
-        assert_almost_equal(eu0[0],1,8)
-        assert_almost_equal(eu2[2],-1,8)
+        assert_allclose(eu0, [1], rtol=1e-15)
+        assert_allclose(eu1, [1, 0], rtol=1e-15)
```

Comment:
> Note that all of the odd Euler numbers are exactly zero

Ah, missed that.

> But in this case all we have to do is check whether the input is odd and if it is return 0, so IMO we should get it exactly right.

Okay, that's what the changed test does. So let's put it in and see if anything falls over.